engine/transaction/draft/create_paymail_draft_test.go

```diff
@@ -0,0 +1,264 @@
+			spec: &outputs.Paymail{
+				To:       "test",
+				Satoshis: transactionSatoshiValue,
+			},
+			expectedError: "paymail address is invalid",
+		},
+		"for domain without alias": {
+			spec: &outputs.Paymail{
+				To:       "@example.com",
+				Satoshis: transactionSatoshiValue,
+			},
+			expectedError: "paymail address is invalid",
+		},
+		"for paymail with invalid alias": {
+			spec: &outputs.Paymail{
+				To:       "$$$@example.com",
+				Satoshis: transactionSatoshiValue,
+			},
+			expectedError: "paymail address is invalid",
+		},
+		"for paymail with invalid domain": {
+			spec: &outputs.Paymail{
+				To:       "test@example.com.$$$",
+				Satoshis: transactionSatoshiValue,
+			},
+			expectedError: "paymail address is invalid",
+		},
+	}
+	for name, test := range errorTests {
+		t.Run("return error "+name, func(t *testing.T) {
+			// given:
+			paymailClient := paymailmock.CreatePaymailClientService(testDomain)
+			paymailClient.WillRespondWithP2PCapabilities()
+
+			// and:
+			draftService := draft.NewDraftService(paymailClient, tester.Logger())
+
+			// and:
+			spec := &draft.TransactionSpec{
+				Outputs: outputs.NewSpecifications(test.spec),
+			}
+
+			// when:
+			tx, err := draftService.Create(context.Background(), spec)
+
+			// then:
+			require.Error(t, err)
+			require.ErrorContains(t, err, test.expectedError)
+			require.Nil(t, tx)
+		})
+	}
+
+	paymailErrorTests := map[string]struct {
+		paymailHostScenario func(*paymailmock.PaymailClientMock)
+		expectedError       string
+	}{
+		"paymail host is responding with not found on capabilities": {
+			paymailHostScenario: func(paymailClient *paymailmock.PaymailClientMock) {
+				paymailClient.WillRespondWithNotFoundOnCapabilities()
+			},
+			expectedError: "paymail host is responding with error",
+		},
+		"paymail host is failing on capabilities": {
+			paymailHostScenario: func(paymailClient *paymailmock.PaymailClientMock) {
+				paymailClient.WillRespondWithErrorOnCapabilities()
+			},
+			expectedError: "paymail host is responding with error",
+		},
+		"paymail host is not supporting p2p destinations capability": {
+			paymailHostScenario: func(paymailClient *paymailmock.PaymailClientMock) {
+				paymailClient.WillRespondWithBasicCapabilities()
+			},
+			expectedError: "paymail host is not supporting P2P capabilities",
+		},
+		"paymail host is failing on p2p destinations": {
+			paymailHostScenario: func(paymailClient *paymailmock.PaymailClientMock) {
+				paymailClient.
+					WillRespondWithP2PCapabilities().
+					WillRespondOnCapability(paymail.BRFCP2PPaymentDestination).
+					WithInternalServerError()
+			},
+			expectedError: "paymail host is responding with error",
+		},
+		"paymail host p2p destinations is returning not found": {
+			paymailHostScenario: func(paymailClient *paymailmock.PaymailClientMock) {
+				paymailClient.
+					WillRespondWithP2PCapabilities().
+					WillRespondOnCapability(paymail.BRFCP2PPaymentDestination).
+					WithNotFound()
+			},
```

Comment:
The `expectedError` field is missing here. Did you do it on purpose?

Reply:
No, my mistake

Reply:
fixed